src/components/menu/menu.test.js

```diff
@@ -0,0 +1,1260 @@
+        getComponent("menu-fullscreen").should(
+          "have.css",
+          "left",
+          `${after}px`
+        );
+        getComponent("menu-fullscreen").should(
+          "have.css",
+          "right",
+          `${after}px`
+        );
+      }
+    );
+
+    it("should verify that the width of the whole submenu with a long-labelled item is determined by the width of the item", () => {
+      CypressMountWithProviders(
+        <Box mb={150}>
+          {["white"].map((menuType) => (
+            <div key={menuType}>
+              <Menu menuType={menuType}>
+                <MenuItem submenu="Menu Item One">
+                  <MenuItem href="#">
+                    Item Submenu One Is A Very Long Submenu Item Indeed
+                  </MenuItem>
+                  <MenuItem variant="alternate" href="#">
+                    Item Submenu Two
+                  </MenuItem>
+                </MenuItem>
+              </Menu>
+            </div>
+          ))}
+        </Box>
+      );
+
+      submenu().eq(positionOfElement("first")).trigger("mouseover");
+      innerMenu(positionOfElement("second"), span).then(($item) => {
+        submenuBlock().should("have.css", "width", `${$item.width()}px`);
+      });
+    });
+
+    it("should verify that the width of the whole submenu with a long title is determined by the width of the title", () => {
+      CypressMountWithProviders(
+        <Box mb={150}>
+          {["white"].map((menuType) => (
+            <div key={menuType}>
+              <Menu menuType={menuType}>
+                <MenuItem submenu="Menu Item One Has A Very Long Menu Title For No Reason Whatsoever">
+                  <MenuItem href="#">Item Submenu One</MenuItem>
+                  <MenuItem variant="alternate" href="#">
+                    Item Submenu Two
+                  </MenuItem>
+                </MenuItem>
+              </Menu>
+            </div>
+          ))}
+        </Box>
+      );
+
+      submenu().eq(positionOfElement("first")).trigger("mouseover");
+      submenu()
+        .eq(positionOfElement("first"))
+        .then(($menu) => {
+          submenuBlock().should("have.css", "width", `${$menu.width()}px`);
+        });
+    });
+
+    it.each([
+      [0.3, 405],
+      [0.6, 810],
+      [1, 1350],
+      [350, 350],
+      [900, 900],
+      [1350, 1350],
+      ["450px", 450],
+      ["675px", 675],
+      ["1200px", 1200],
+    ])(
+      "should verify Menu width is set by width prop when passed as a decimal, an integer or a string",
+      (width, pixels) => {
+        CypressMountWithProviders(<MenuComponent width={width} />);
+
+        menu().should("have.css", "width", `${pixels}px`);
+      }
+    );
+
+    it.each([
+      [15, 15],
+      [27, 27],
+      [41, 41],
+      ["10px", 10],
+      ["30px", 30],
+      ["50px", 50],
+    ])(
+      "should verify Menu height is set by height prop when passed as an integer or a string",
```

Comment:
same here for a type, we can actually wrap all these tests where we are passing this sentence `when passed as an integer or a string`